drivers/watchdog/wdt_nrfx.c

```diff
@@ -73,6 +73,12 @@ static int wdt_nrf_disable(const struct device *dev)
 		return -EFAULT;
 	}
 
+#if !CONFIG_WDT_NRFX_NO_IRQ
+	while (!nrfy_wdt_events_process(config->wdt.p_reg,
```

Comment:
if `CONFIG_WDT_NRFX_NO_IRQ` is not set (default setting), nrfx WDT IRQ handler is installed and will execute once `NRF_WDT_EVENT_STOPPED`. `NRF_WDT_EVENT_STOPPED` will be cleared in IRQ context, so this busy loop might execute forever.

Instead of polling event, `k_sem_take` should be called here. Corresponding `k_sem_give` should be executed in `wdt_event_handler` upon reception on `NRF_WDT_EVENT_STOPPED`.

Also, not all nRF devices can stop WDT so this extra code in `wdt_event_handler` needs to be enclosed in `#if NRFX_WDT_HAS_STOP`